### DISCUSSION TITLE
Feature request: Update docs for connectHits to support preferred highlighting method

### DISCUSSION THREAD

**Discussion by sgilberg** [algolia/instantsearch#6408] [Ideas] [answered]
The documentation for customizing the UI with the `connectHits()` function still uses `$(instantsearch.highlight({ attribute: 'name', hit: item }))` but this yields a deprecation warning. However the `components` feature doesn't appear to be available to this method in the same way that it is when defining templates in [autocomplete](https://www.algolia.com/doc/ui-libraries/autocomplete/api-reference/autocomplete-js/autocomplete/#components) for example . Is there a non-deprecated way to highlight within `connectHits()`, and if so, can the documentation be updated to reflect this?

**Comment by Haroenv** [comment, marked as answer]
Good point, this is indeed not yet documented. I'm finding a place to document it now. In the mean time you can already look at how the example using connectInfiniteHits is changed here: https://github.com/algolia/instantsearch/pull/6410 (code in a clean way visible here: https://github.com/algolia/instantsearch/blob/1f0bc01b598ac9ce613d68f7dcc595f8f03b3744/examples/js/media/src/widgets/Articles.ts)

The previous highlight and snippet helper never were safe as they were still passed to innerHTML. Using htm as inside InstantSearch works much better!

Hope that helps!

**Reply by sgilberg** [reply]
Thanks @Haroenv! That got me pointed in the right direction to implement something similar importing `render`, `html`, and `Highlight`.
